src/test/java/carparks/resources/ParkingEntryControllerShould.java

```diff
@@ -0,0 +1,129 @@
+package carparks.resources;
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.math.BigDecimal;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+
+import com.google.gson.Gson;
+
+import carparks.parking.DailySummaryDto;
+import carparks.parking.FeeType;
+import carparks.parking.ParkingEntryDto;
+import carparks.parking.ParkingService;
+import carparks.parking.SessionStatus;
+
+
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(ParkingEntryController.class)
+public class ParkingEntryControllerShould {
+
+	@Autowired
+    private MockMvc mockMvc;
+	
+	@MockBean
+	private ParkingService service;
+	
+	@Test
+	public void return_parking_entry_with_get_by_plate_number() throws Exception {
+		
+		// given
+		ParkingEntryDto entry = new ParkingEntryDto("WF12345", new BigDecimal(12), SessionStatus.FINISHED);
+		when(service.getEntry(anyString(), anyString())).thenReturn(entry);
+		
+		// when & then
+		this.mockMvc.perform(get("/platenumber/WF12345")).andExpect(status().isOk())
+        .andExpect(content().json("{'plateNumber':'WF12345','fee':12.00,'status':'FINISHED'}"));
```

Comment:
A co jeśli zmieniłbyś formatowanie jsona? Albo jakbyś dodał nowe pole np. `timestamp`? Czy widzisz w tym problem, że przy każdej takiej zmianie trzeba przerabiać testy? Jak można by to inaczej rozwiązać?

Reply:
Ok, chyba rozumiem problem - chodzi o to, żeby test sprawdzał czy json zawiera jakąś wartość a nie dokładnie jak ten json wygląda, bo teraz to nawet zmiana kolejności wartości wywali test. Można to poprawić np używając jsonPath.

Reply:
Tak, takie podejście sprawiłoby, że testy byłby mniej zabetonowane